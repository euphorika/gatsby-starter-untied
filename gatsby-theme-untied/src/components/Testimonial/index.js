/** @jsx jsx */
import { jsx } from "theme-ui"
import PropTypes from "prop-types"

import Img from "gatsby-image"

const Testimonial = ({ children, name, imgFixed }) => (
  <section
    sx={{
      p: 3,
    }}
  >
    <div
      sx={{
        textAlign: "center",
      }}
    >
      <Img fixed={imgFixed} />
    </div>
    <blockquote
      sx={{
        my: 3,
        mx: "auto",
        fontSize: 0,
        maxWidth: "400px",
        textAlign: "center",
        "&::before": {
          content: "open-quote",
        },
        "&::after": {
          content: "close-quote",
        },
      }}
    >
      {children}
    </blockquote>
    <div
      sx={{
        textAlign: "center",
      }}
    >
      <strong>{name}</strong>
    </div>
  </section>
)

Testimonial.propTypes = {
  children: PropTypes.node.isRequired,
  name: PropTypes.string.isRequired,
  imgFixed: PropTypes.object.isRequired,
}

export default Testimonial
