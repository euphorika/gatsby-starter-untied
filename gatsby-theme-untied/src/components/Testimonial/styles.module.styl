@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

.testimonial-section
  padding: 15px

  blockquote
    margin: 15px auto
    font-size: 12px
    max-width: 400px
    text-align center
    &::before
      content: open-quote
    &::after
      content: close-quote

  .author-img
    text-align center

  .author
    text-align center
