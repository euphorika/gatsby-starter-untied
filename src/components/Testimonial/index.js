import React from 'react'

import styles from './styles.module.styl'

const Testimonial = () => (
  <section className={styles.testimonialSection}>Testimonial</section>
)

export default Testimonial
