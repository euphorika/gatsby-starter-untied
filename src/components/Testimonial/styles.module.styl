@require '../../themes/standard/config/breakpoints'
@require '../../themes/standard/config/colors'

.testimonial-section
  margin 0
