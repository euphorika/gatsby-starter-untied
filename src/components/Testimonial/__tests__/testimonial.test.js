import React from 'React'
import renderer from 'react-test-renderer'
import Testimonial from '../'

describe('Testimonial', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Testimonial />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
