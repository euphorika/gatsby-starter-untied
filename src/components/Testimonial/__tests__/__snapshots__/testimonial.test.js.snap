// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonial renders correctly 1`] = `
<section
  className="testimonialSection"
>
  Testimonial
</section>
`;
